jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -6616,36 +6620,204 @@ private static char convEndian(boolean isBigEndian, char value) {
 	 */
 	public native long getObjectSize(Object o);
 
+	/**
+	 * Determines whether a field at a given offset in a given class is flattened
+	 *
+	 * @param clz the the class to check
+	 * @param offset offset in bytes of the field in clz
+	 * @return a boolean indicating whether a flattened field exists at the given offset
+	 */
+	private native boolean isFieldAtOffsetFlattened(Class<?> clz, long offset);
+
+	/**
+	 * Sets the primitive type at offset in obj if the parameter v1
+	 * matches the existing value in the object.
+	 *
+	 * @param obj object into which to store the value
+	 * @param offset location to compare and store value in obj
+	 * @param clz the class of the parameters v1 and v2
+	 * @param v1 value that is expected to be in obj at offset
+	 * @param v2 value that will be set in obj at offset if compare is successful
```

Comment:
"if compare is successful" should be "if the comparison is successful". (I see "if compare is successful" in multiple places in this file, please fix them all)

Reply:
changed in 7159341